### ISSUE TITLE
Cannot compile on other than latest as action is only doing a shallow copy of micropython

### ISSUE THREAD

**Issue by brendancolevalley** [BrianPugh/install-micropython#9]
Thanks for creating the action - its helpful. I have run into an issue that prevents using older versions of micropython as it is doing a shallow copy by default.  In the javascript, the code to clone the micropython repository is:

```javascript
  // Shallow clone the repository to get the latest commit hash or checkout the provided reference
  await exec.exec(`git clone --depth 1 ${repository} ${mpy_dir}`);

  if(reference){
    await exec.exec(`git checkout ${reference}`, [], {cwd: mpy_dir});
  }
```
  
It probably should be something like:
```javascript
if (reference) {
  // If a reference is provided, clone the specific tag with a shallow copy
  await exec.exec(`git clone --branch ${reference} --depth 1 ${repository} ${mpy_dir}`);
} else {
  // If no reference is provided, perform a shallow clone of the repository to get the latest commit
  await exec.exec(`git clone --depth 1 ${repository} ${mpy_dir}`);
}
```

**Comment by BrianPugh** [comment]
`reference` could be a tag or a commit hash, which I don't think would work with that change. I'm kind of a javascript noob, so I'd highly appreciate a PR!

**Comment by brendancolevalley** [comment]
Fellow js noob here! I am also a beginner with git :-) 
You are correct, what I provided will not work for a commit hash, it just works for branch or tag name. But maybe that's ok? My guess is that most use cases would be for major versions?
I think if you want to support commit hash, it involves cloning the full repository and then checking out the commit hash which adds some complexity. 
Let me know how I can best help. Happy to create a PR/ test if you want to leverage the approach I provided using name, I just want to make sure you are aware of my experience level.....

**Comment by BrianPugh** [comment]
while not terribly scientific, should I just get rid of the `--depth 1`? Running locally:

```
$ time git clone git@github.com:micropython/micropython.git
Cloning into 'micropython'...
remote: Enumerating objects: 117248, done.
remote: Counting objects: 100% (223/223), done.
remote: Compressing objects: 100% (141/141), done.
remote: Total 117248 (delta 101), reused 125 (delta 82), pack-reused 117025
Receiving objects: 100% (117248/117248), 59.95 MiB | 17.20 MiB/s, done.
Resolving deltas: 100% (87501/87501), done.
git clone git@github.com:micropython/micropython.git  4.14s user 0.89s system 75% cpu 6.624 total
```

```
$ time git clone --depth 1 git@github.com:micropython/micropython.git
Cloning into 'micropython'...
remote: Enumerating objects: 5619, done.
remote: Counting objects: 100% (5619/5619), done.
remote: Compressing objects: 100% (4555/4555), done.
remote: Total 5619 (delta 1341), reused 3126 (delta 739), pack-reused 0
Receiving objects: 100% (5619/5619), 8.01 MiB | 2.25 MiB/s, done.
Resolving deltas: 100% (1341/1341), done.
git clone --depth 1 git@github.com:micropython/micropython.git  0.61s user 0.52s system 18% cpu 6.146 total
```

It seems like specifying the depth takes around the same amount of time (6.146s vs 6.624s), but reduces the bandwidth from 60MB to 8.01MB. I don't think the savings is significant enough for anyone to really care.

**Comment by brendancolevalley** [comment]
I think that should do, but I will provide a pull request so ` --depth 1` can be kept if reference is not specified. I'll be able to test it to make sure it works for MP 1.20 and latest.

**Comment by BrianPugh** [comment]
#10 has been merged and released as v1.0.6. Thank you!
